src/functions-and-arrays.js

```diff
@@ -1,38 +1,101 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(num1,num2) {
+ if(num1 > num2){
+ return num1;
+} 
+  else if (num1 < num2) {
+  return num2;
+}
+  else   {
+    return (num1, num2);
+}
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+function findLongestWord(words){
+if (!words.length){
+  return null;
+}
+
+let longestWord ="";
+
+
+words.forEach(function(word){
+  if(word.length > longestWord.length) {
+
+  longestWord = word;
+  }
+});
+
+return longestWord;
 
-function findLongestWord() {}
+
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0;
+  numbers.forEach(lab => {
+      sum += lab;
+  });
+  console.log(sum);
+ return sum;
+
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(numbers) {
+  let sum = 0;
+  numbers.forEach(lab =>{
+     sum += lab;
+  });
+  console.log(sum);
+  return sum;
+
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbers) {
+  if(numbers.length === 0){
+    return null;
+  }
+let sum = 0;
```

Comment:
Since you have a function that you wrote earlier that sums numbers, you can reuse it here.
Something like this:
```
function averageNumbers(numbers) {
  if (numbers.length === 0) return null;
  let sum = sumNumbers(numbers);
  let average = sum / numbers.length;
  return average;
}
```